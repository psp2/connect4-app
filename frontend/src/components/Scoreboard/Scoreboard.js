import React from "react";
import "./Scoreboard.scss";

function Scoreboard(props) {
  // Temp until API, repeated games implemented

  return (
    <div className="Scoreboard">
      <div className="ScoreboardTitle">Scoreboard</div>
      <hr className="ScoreboardHRLine"></hr>
      <div className="ScoreboardTable">
        <div className="ScoreboardNameColumn">
          <div className="ScoreboardName">{props.player1Name}</div>
          <div className="ScoreboardName">{props.player2Name}</div>
        </div>
        <vr className="ScoreboardVRLine"></vr>
        <div className="ScoreboardNameColumn">
          <div className="ScoreboardName">{props.player1Score}</div>
          <div className="ScoreboardName">{props.player2Score}</div>
        </div>
      </div>
    </div>
  );
}

export default Scoreboard;
