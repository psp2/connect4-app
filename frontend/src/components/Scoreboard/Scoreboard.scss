.Scoreboard {
  margin: 5vh;
  height: 55vh;
  width: 60vw;
  background-color: #106eeb;
  border-radius: 20px;
}

.ScoreboardTitle {
  color: white;
  font-size: 5em;
  font-family: "Roboto", sans-serif;
  margin: auto;
}

.ScoreboardTable {
  display: flex;
  flex-direction: row;
}

.ScoreboardNameColumn {
  display: flex;
  flex-direction: column;
  height: 32vh;
  margin: 5vh;
  margin: auto;
}

.ScoreboardName {
  color: white;
  font-size: 5em;
  font-family: "Roboto", sans-serif;
  margin: auto;
}

.ScoreboardHRLine {
  margin-top: 2vh;
  border: 2px solid white;
}

.ScoreboardVRLine {
  margin-top: -1.3vh;
  height: 41vh;
  border: 2px solid white;
}
