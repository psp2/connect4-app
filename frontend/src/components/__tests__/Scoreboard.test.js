import React from "react";
import Scoreboard from "../Scoreboard/Scoreboard";
import renderer from "react-test-renderer";

it("Scoreboard1, renders correctly", () => {
  const player1Name = "Prashant";
  const player1Score = 3;
  const player2Name = "Vivek";
  const player2Score = 2;
  const tree = renderer
    .create(
      <Scoreboard
        player1Name={player1Name}
        player1Score={player1Score}
        player2Name={player2Name}
        player2Score={player2Score}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Scoreboard2, renders correctly", () => {
  const player1Name = "Vivek";
  const player1Score = 3;
  const player2Name = "Prashant";
  const player2Score = 2;
  const tree = renderer
    .create(
      <Scoreboard
        player1Name={player1Name}
        player1Score={player1Score}
        player2Name={player2Name}
        player2Score={player2Score}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Scoreboard, null values", () => {
  const player1Name = null;
  const player1Score = null;
  const player2Name = null;
  const player2Score = null;
  const tree = renderer
    .create(
      <Scoreboard
        player1Name={player1Name}
        player1Score={player1Score}
        player2Name={player2Name}
        player2Score={player2Score}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
