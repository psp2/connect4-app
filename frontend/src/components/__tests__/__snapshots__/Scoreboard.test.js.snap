// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scoreboard, null values 1`] = `
<div
  className="Scoreboard"
>
  <div
    className="ScoreboardTitle"
  >
    Scoreboard
  </div>
  <hr
    className="ScoreboardHRLine"
  />
  <div
    className="ScoreboardTable"
  >
    <div
      className="ScoreboardNameColumn"
    >
      <div
        className="ScoreboardName"
      />
      <div
        className="ScoreboardName"
      />
    </div>
    <vr
      className="ScoreboardVRLine"
    />
    <div
      className="ScoreboardNameColumn"
    >
      <div
        className="ScoreboardName"
      />
      <div
        className="ScoreboardName"
      />
    </div>
  </div>
</div>
`;

exports[`Scoreboard1, renders correctly 1`] = `
<div
  className="Scoreboard"
>
  <div
    className="ScoreboardTitle"
  >
    Scoreboard
  </div>
  <hr
    className="ScoreboardHRLine"
  />
  <div
    className="ScoreboardTable"
  >
    <div
      className="ScoreboardNameColumn"
    >
      <div
        className="ScoreboardName"
      >
        Prashant
      </div>
      <div
        className="ScoreboardName"
      >
        Vivek
      </div>
    </div>
    <vr
      className="ScoreboardVRLine"
    />
    <div
      className="ScoreboardNameColumn"
    >
      <div
        className="ScoreboardName"
      >
        3
      </div>
      <div
        className="ScoreboardName"
      >
        2
      </div>
    </div>
  </div>
</div>
`;

exports[`Scoreboard2, renders correctly 1`] = `
<div
  className="Scoreboard"
>
  <div
    className="ScoreboardTitle"
  >
    Scoreboard
  </div>
  <hr
    className="ScoreboardHRLine"
  />
  <div
    className="ScoreboardTable"
  >
    <div
      className="ScoreboardNameColumn"
    >
      <div
        className="ScoreboardName"
      >
        Vivek
      </div>
      <div
        className="ScoreboardName"
      >
        Prashant
      </div>
    </div>
    <vr
      className="ScoreboardVRLine"
    />
    <div
      className="ScoreboardNameColumn"
    >
      <div
        className="ScoreboardName"
      >
        3
      </div>
      <div
        className="ScoreboardName"
      >
        2
      </div>
    </div>
  </div>
</div>
`;
